test_runner/regress/test_lsn_mapping.py

```diff
@@ -108,6 +109,43 @@ def test_lsn_mapping(neon_env_builder: NeonEnvBuilder):
         assert Lsn(result["lsn"]) >= last_flush_lsn
 
 
+#
+# Test if cancelled pageserver get_lsn_by_timestamp request is correctly handled.
+#
+# Added as an effort to improve error handling and avoid full anyhow backtrace
+# https://github.com/neondatabase/neon/pull/7949
+#
+def test_get_lsn_by_timestamp_cancelled(neon_env_builder: NeonEnvBuilder):
+    env = neon_env_builder.init_start()
+    endpoint = env.endpoints.create_start("main")
+
+    tenant_id = env.initial_tenant
+    timeline_id = env.initial_timeline
+
+    conn = endpoint.connect()
+    cur = conn.cursor()
+    cur.execute("CREATE TABLE foo (x integer)")
+    tbl = []
+    cur.execute("INSERT INTO foo VALUES(0)")
+    # Get the timestamp at UTC
+    after_timestamp = query_scalar(cur, "SELECT clock_timestamp()").replace(tzinfo=None)
+    tbl.append([0, after_timestamp])
+    cur.execute("INSERT INTO foo VALUES (-1)")
+
+    with env.pageserver.http_client() as client:
+        failpoint = "timeline-request-cancelled"
+        client.configure_failpoints((failpoint, "return"))
+
+        with pytest.raises(
+            PageserverApiException,
+            match="Request cancelled",
+        ) as exc:
+            probe_timestamp = tbl[0][1]
+            client.timeline_get_lsn_by_timestamp(tenant_id, timeline_id, probe_timestamp)
```

Comment:
Thinking this again, I don't think this will actually reproduce the original case in the issue.

Original case in the issue was a logging "Cancelled request failed: {e:?}" or something like that.

This will only hit the failpoint, but it will not exercise the same path. In the original case, the requester was not around to wait for the response which is inspected here.

Reply:
One could argue we shouldn't be logging it as an `error!` when this new Cancelled variant is observed post cancellation.